src/main/java/com/mateacademy/util/SecurityUtil.java

```diff
@@ -0,0 +1,44 @@
+package com.mateacademy.util;
+
+import com.mateacademy.model.Role;
+import com.mateacademy.model.UserModel;
+import com.mateacademy.service.UserService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.stereotype.Component;
+
+import java.security.Principal;
+import java.util.Objects;
+import java.util.Optional;
+
+@Component
+@Slf4j
+public class SecurityUtil {
```

Comment:
https://www.vojtechruzicka.com/avoid-utility-classes/
It is not utility class. Please rename it.
SecurityUtil ->SecurityService for example